LibRangeCheck-3.0/LibRangeCheck-3.0.lua

```diff
@@ -771,11 +771,7 @@ local function getRange(unit, noItems)
   end
 
   if UnitCanAttack("player", unit) then
-    if InCombatLockdownRestriction(unit, true) then
-      return getRangeWithCheckerList(unit, noItems and lib.harmNoItemsRCInCombat or lib.harmRCInCombat)
-    else
-      return getRangeWithCheckerList(unit, noItems and lib.harmNoItemsRC or lib.harmRC)
-    end
+    return getRangeWithCheckerList(unit, noItems and lib.harmNoItemsRC or lib.harmRC)
```

Comment:
So the issue of double checking UnitCanAttack also exists for InCombatLockdownRestriction(). 

The library maintains two lists for each unit type, one e.g. harmRC vs harmRCInCombat. The difference being that harmRCInCombat doesn't contain any restricted checkers. Here and every other call to getRangeWithCheckerList InCombatLockdownRestriction is used to decide which list to use. 

Thus the checkers in those lists don't actually have to do their own InComatLockdownRestriction check again.

The reason the checker do check again is that in addition to the "check range of unit" apis, LibRangeCheck also provides apis that return a list of checkers.

My change does the following:
* Since the list we choose only contains checkers that work, while called from the library the InCombatLockdownRestriction check can be skipped.
* For library users that get a list of checkers, each checker checks the InCombatLockdownRestriction.